lms/static/scripts/frontend_apps/components/AutoGradingConfigurator.tsx

```diff
@@ -0,0 +1,208 @@
+        step={1}
+        value={value}
+        onChange={e => onChange(Number((e.target as HTMLInputElement).value))}
+      />
+    </div>
+  );
+}
+
+export type AutoGradingConfiguratorProps = {
+  autoGradingConfig: AutoGradingConfig;
+  updateAutoGradingConfig: (newConfig: AutoGradingConfig) => void;
+};
+
+/**
+ * Allows instructors to enable auto grading for an assignment, and provide the
+ * configuration to determine how to calculate every student's grade.
+ */
+export default function AutoGradingConfigurator({
+  autoGradingConfig,
+  updateAutoGradingConfig,
+}: AutoGradingConfiguratorProps) {
+  const {
+    active = false,
+    gradingType,
+    activityCalculation,
+    requiredAnnotations,
+    requiredReplies = 1,
+  } = autoGradingConfig;
+  const updatePartialConfig = useCallback(
+    (partialConfig: Partial<AutoGradingConfig>) =>
+      updateAutoGradingConfig({ ...autoGradingConfig, ...partialConfig }),
+    [autoGradingConfig, updateAutoGradingConfig],
+  );
+
+  const gradingTypeId = useId();
+  const activityCalculationId = useId();
+
+  return (
+    <div className="flex flex-col gap-y-3">
+      <Checkbox
+        checked={active}
+        checkedIcon={CheckboxCheckedFilledIcon}
+        onChange={e =>
+          updatePartialConfig({
+            active: (e.target as HTMLInputElement).checked,
+          })
+        }
+      >
+        Enabled automatic participation grading
+      </Checkbox>
+      {active && (
+        <>
+          <div>
+            <p id={gradingTypeId} className="font-semibold mb-1">
+              Grading type
+            </p>
+            <RadioGroup
+              data-testid="grading-type-radio-group"
+              aria-labelledby={gradingTypeId}
+              selected={gradingType}
+              onChange={gradingType => updatePartialConfig({ gradingType })}
+            >
+              <RadioGroup.Radio
+                value="all_or_nothing"
+                subtitle={<small>Must meet minimum requirements.</small>}
+              >
+                All or nothing
+              </RadioGroup.Radio>
+              <RadioGroup.Radio
+                value="scaled"
+                subtitle={<small>Proportional to percent completed.</small>}
+              >
+                Scaled
+              </RadioGroup.Radio>
+            </RadioGroup>
+          </div>
+          <div>
+            <p id={activityCalculationId} className="font-semibold mb-1">
+              Activity calculation
+            </p>
+            <RadioGroup
+              data-testid="activity-calculation-radio-group"
+              aria-labelledby={activityCalculationId}
+              selected={activityCalculation}
+              onChange={activityCalculation =>
+                updatePartialConfig({ activityCalculation })
+              }
+            >
+              <RadioGroup.Radio
+                value="cumulative"
+                subtitle={<small>Annotation & replies tallied together.</small>}
```

Comment:
```suggestion
                subtitle={<small>Annotation and replies tallied together.</small>}
```

We always use the word "and" rather than a symbol in other UI text I believe.